### PR TITLE
Fix `vllm:prompt_tokens_total` metric calculation

### PR DESCRIPTION
**Pull request by ronensc**
I have identified an issue with the `vllm:prompt_tokens_total` counter metric when there are multiple prompts in a batch with different token lengths.

The root cause of the problem lies in the metric counting the token length of the longest prompt in the batch multiplied by the number of prompts in the batch (as if the shorter prompts were padded to match the logest).
Code ref:
https://github.com/vllm-project/vllm/blob/7e45107f51bcb38c22dd9916c61226078e8eb26d/vllm/core/scheduler.py#L262-L263

This PR resolves this issue by accurately counting the tokens of all the prompts in the batch. A simple unit test has been added to validate the correctness of the counter.
Additionally, I have fixed `_read_prompts()` to read all prompts from a file, rather than the first prompt.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -14,11 +14,9 @@
 
 
 def _read_prompts(filename: str) -> str:
```
Comment (NikolaBorisov) [inline tests/conftest.py:16]: Looks like this function has wrong signature. output seems to be List[str]

**Review by NikolaBorisov — APPROVED** [review]
LGTM

File: tests/conftest.py (hunk 1 above)
Comment (ronensc) [reply tests/conftest.py]: Good catch! fixed.

**Comment by NikolaBorisov** [comment]
@simon-mo This is small fix for stats. Safe to merge

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thank you for your contribution!

**Comment by ronensc** [comment]
I noticed that the 2nd prompt of [`prompts/example.txt`](https://github.com/vllm-project/vllm/blob/e433c115bce2bf27f7b1abdde7029566007d9eee/tests/prompts/example.txt#L2) causes some tests to fail.
https://buildkite.com/vllm/ci/builds/1399#018dc04b-091b-46e8-8ac4-6973f0536b82/51-1384


Possibly related to: #975

**Pull request merged**